src/DynamoCoreWpf/Controls/ZoomBorder.cs

```diff
@@ -8,15 +8,34 @@
 using Dynamo.Models;
 using Dynamo.Utilities;
 using Dynamo.ViewModels;
+using System;
 
 namespace Dynamo.Controls
 {
+    class ViewSettingsChangedEventArgs : EventArgs
+    {
+        internal ViewSettingsChangedEventArgs(double x, double y, double zoom)
+        {
+            X = x;
+            Y = y;
+            Zoom = zoom;
+        }
+
+        internal double X { get; private set; }
+        internal double Y { get; private set; }
+        internal double Zoom { get; private set; }
+    }
+
+    internal delegate void ViewSettingsChangedHandler(ViewSettingsChangedEventArgs args);
+
```

Comment:
Every time a zoom/pan happens, it ends up in `ZoomBorder`, which in turns sends out notification through `ViewSettingsChangedHandler` event. `InfiniteGridView` subscribes to this event and reliably updates its visual.